### PR TITLE
UI: Fixes in source view and wizard

### LINKED ISSUE #827: UI: Add Source Wizard issues
## Type:
- Bug

### Bug severity:
- Medium
___

## Description:
A couple of issues with the Add Source Wizard:

- Second page should auto focus on the source name field
- After adding a credential, the credential does not appear in the list
- Selecting a credential doesn't show it as selected when I bring up the list again. It does if I select another credential. But then that one doesn't show up.
- The last page of the wizard says I will get notified, I never do. We should probably just refresh the sources page on an add, it shows up then.

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
Implement Delete source
Update sources list after Add, Edit, Delete
Do no lost updates in wizard when a new credential is added

Fixes #827

### CODE REVIEW

File: client/src/components/addSourceWizard/addSourceWizardStepTwo.js [hunk 1]
```diff
@@ -75,19 +75,31 @@ class AddSourceWizardStepTwo extends React.Component {
       remappedCredentials = credentials.map(val => val.id);
     }
 
+    let udpatedSource;
+    if (nextProps.source !== _.get(this.state, 'source')) {
```
Comment (cdcabrera) [inline client/src/components/addSourceWizard/addSourceWizardStepTwo.js]: There may be some limitations around this check, think using lodash's `_.isEqual(a, b)` might be better

File: client/src/constants/apiConstants.js [hunk 2]
```diff
@@ -16,6 +16,6 @@ export {
   API_SOURCE_SAT_VERSION,
   API_SOURCE_SSL_CERT,
   API_SOURCE_SSL_DISABLE,
-  API_SOURCE_SSL_PROT,
+  API_SOURCE_SSL_PORT,
```
Comment (cdcabrera) [inline client/src/constants/apiConstants.js]: yeahhhh, I should have named this one better, it was actually "prot" not "port"... it's an enum based on the API swagger doc...  `SSLv2, SSLv23, SSLv3, TLSv1, TLSv1_1, TLSv1_2`... maybe flip it to `PROTOCOL`

Comment (jeff-phillips-18) [reply client/src/constants/apiConstants.js]: ah....

File: client/src/components/addSourceWizard/addSourceWizardStepTwo.js (hunk 1 above)
Comment (jeff-phillips-18) [reply client/src/components/addSourceWizard/addSourceWizardStepTwo.js]: I'll update to use _.isEqual for now. I think we need a deeper dive on the use of props and state for this component.

File: client/src/components/addSourceWizard/addSourceWizardStepTwo.js [hunk 3]
```diff
@@ -125,19 +122,22 @@ class AddSourceWizardStepTwo extends React.Component {
...
-
+      let updatedSource = { id: source.id };
+      _.set(updatedSource, apiTypes.API_SOURCE_TYPE, sourceType);
+      _.set(updatedSource, apiTypes.API_SOURCE_NAME, sourceName);
+      _.set(updatedSource, apiTypes.API_SOURCE_HOSTS, hosts);
+      _.set(updatedSource, apiTypes.API_SOURCE_SAT_VERSION, satelliteVersion);
+      _.set(updatedSource, apiTypes.API_SOURCE_SSL_PROTOCOL, sslProtocol);
+      _.set(updatedSource, apiTypes.API_SOURCE_SSL_CERT, sslCertVerify);
+      _.set(updatedSource, apiTypes.API_SOURCE_SSL_DISABLE, disableSsl);
+      _.set(updatedSource, apiTypes.API_SOURCE_CREDENTIALS, credentials);
```
Comment (cdcabrera) [inline client/src/components/addSourceWizard/addSourceWizardStepTwo.js]: This partially occurred to me before, viewing this update I've had a reoccurrence... I think we can definitely look at focusing the steps so they only update the properties they're responsible for, similar to using the former `_.merge` that was on `line 140`. If we handle it that way we might help alleviate some muddled confusion around 

> Why are we aiming to re-map the source object, both in resetting the state and in passing it back to Redux... what if we narrowed the scope even more, or at least until step 2 ends up handling the remaining properties. That ties into only having to observe each prior steps updated properties, similar to the "step valid" property.

I think if we combined this concept, with some form of a validation component that simply returns a value towards the state source object/model, and internally handles the validation, we could cut the wizard steps down even further.

Comment (jeff-phillips-18) [reply client/src/components/addSourceWizard/addSourceWizardStepTwo.js]: Yeah, basically what this is doing. I can remove the fields that are not updated by this page and merge instead.

**Comment by jeff-phillips-18** [comment]
Added the /, thanks!

**Pull request merged**